src/parallel/MPI_TIMED/ProcessTimer.h

```diff
@@ -0,0 +1,120 @@
+/*
+ * ProcessTimer.h
+ *
+ *  Created on: Apr 12, 2020
+ *      Author: jeremyharisch
+ */
+
+#ifndef MARDYN_PROCESSTIMER_H
+#define MARDYN_PROCESSTIMER_H
+#define _BSD_SOURCE
+
+#include <fstream>
+#include <vector>
+#include <array>
+#include <map>
+#include <sys/time.h>
+
+#include <mpi.h>
+
+
+class ProcessTimer {
+
+public:
+	ProcessTimer() {}
+
+	~ProcessTimer() {}
+
+	/*! @brief Switches between the profiling Levels
+	 *  This Fnction should be called right before and after a corresponding MPI-Function-Call
+	 *  to exclude only this one
+	 *
+	 *  @param level ==1 Profiling is enabled(default); ==0 Profiling is disabled
+	 */
+	int switchProfiling(const int level = 1) {
+		_profiling_switch = level;
+		return _profiling_switch;
+	}
+
+	//! @brief Starts Timer for MPI-Measurement
+	//!
+	//! @param process Rank of process
+	void startTimer() {
+		if (_profiling_switch == 1){
+			int process;
+			MPI_Comm_rank(MPI_COMM_WORLD, &process);
+			double measurement_time = MPI_Wtime();
+			_process_time -= measurement_time;
+			_processes_debug[process].push_back(-measurement_time);
```

Comment:
do you need to do this push_back? can't you instead open the log at the first use of the timer? closing of the ofstream would then be automatic, at the destructor

Reply:
This map is used to save all the measured times in ProcessTimer for better debugging. This pushback inserts the new started time into the map, which then, later on, can be printed to a file. If I would directly write it to the file it may take more time, because when the timer gets stopped this time need to be read and than subtracted and write again into it.

Reply:
do you actually need the map? `process` should be always the same, shouldn't it?
I'm fine with the vector, but the map seems unneeded :)

Reply:
With the map, I am also saving the process rank for better debugging. I think this is the easiest solution to store the rank including its time

Reply:
you don't need to store it at all, you can always query it with MPI_Comm_rank...